lib/experiment-prompt.js

```diff
@@ -0,0 +1,133 @@
+const etch = require('etch')
+const $ = etch.dom
+const SCORING_SYSTEMS = require('./scoring-systems')
+
+/**
+ * For how long to show the "NEW" badge in the prompt since the first time
+ * it was seen (24h).
+ */
+const TimeToShowNewBadgeInMilliseconds = 24 * 60 * 60 * 1000
+const FirstTimeShownKey = 'fuzzy-finder:prompt-first-time-shown'
+
+function shouldShowBadge () {
+  const firstTimeShown = localStorage.getItem(FirstTimeShownKey)
+  const now = new Date().getTime()
+
+  if (!firstTimeShown) {
+    localStorage.setItem(FirstTimeShownKey, now)
+
+    return true
+  }
+
+  return now - firstTimeShown <= TimeToShowNewBadgeInMilliseconds
+}
+
+function showNotification (message, {description, confirmBtn, cancelBtn, confirmFn, cancelFn}) {
+  const notification = atom.notifications.addInfo(
+    message,
+    {
+      description,
+      dismissable: true,
+      buttons: [
+        {
+          text: confirmBtn,
+          onDidClick: () => {
+            confirmFn && confirmFn()
+            notification.dismiss()
+          },
+          className: 'btn btn-info btn-primary'
+        },
+        {
+          text: cancelBtn,
+          onDidClick: () => {
+            cancelFn && cancelFn()
+            notification.dismiss()
+          }
+        }
+      ],
+      icon: 'rocket'
+    }
+  )
+}
+
+function enableExperimentalFuzzyFinder (metricsReporter) {
+  metricsReporter.incrementCounter('click-enable-prompt')
+
+  showNotification(
+    'Introducing experimental fast mode',
+    {
+      description: "The fuzzy finder has a new experimental _fast mode_. It dramatically speeds up the experience of finding files, especially in large projects. Would you like to try it?\n\n(You can always switch back to the fuzzy finder's _normal mode_ later.)",
+      confirmBtn: 'Enable fast mode',
+      confirmFn: () => {
+        metricsReporter.incrementCounter('confirm-enable-prompt')
+
+        atom.config.set('fuzzy-finder.useRipGrep', true)
+        atom.config.set('fuzzy-finder.scoringSystem', SCORING_SYSTEMS.FAST)
+      },
+      cancelBtn: 'Not right now',
+      cancelFn: () => {
+        metricsReporter.incrementCounter('cancel-enable-prompt')
+      }
+    }
+  )
+}
+
+function disableExperimentalFuzzyFinder (metricsReporter) {
+  metricsReporter.incrementCounter('click-disable-prompt')
+
+  showNotification(
+    'Do you want to disable the new experimental fast mode for the fuzzy finder?',
+    {
+      description: 'If the experimental fast mode is negatively impacting your experience, please [**let us know**](https://github.com/atom/fuzzy-finder/issues/new?title=[Feedback]+Experimental+fast+mode).\n\n(You can reenable fast mode later from the fuzzy finder.)',
```

Comment:
@rafeca: 3bc0884c592e95af179249d314e242a045165533 encourages users to let us know if they're having issues with the experimental fast mode:

<img width="482" alt="Screen Shot 2019-04-09 at 7 34 25 PM" src="https://user-images.githubusercontent.com/2988/55841770-99cc3200-5afe-11e9-8bf1-9a7729622a46.png">

Currently, following that link will open the form for creating a new issue:

https://github.com/atom/fuzzy-finder/issues/new?title=[Feedback]+Experimental+fast+mode

Alternatively, we could start a thread on https://discuss.atom.io/, include a link to that thread, and encourage users to post a comment describing any any problems they encounter. That might help us keep the GitHub issues a bit tidier. What do you think?

Reply:
I think I prefer the idea of using a single place were people can comment, this way we don't have to handle all the issues separately and it'll be easier to group the feedback.

Also, instead of using a discuss thread we can create a "master issue" on GitHub for feedback, so people can comment on the issue the same way they would do on discuss: for me it would be (slightly) more natural to monitor a GitHub issue than a thread on discuss.

Anyways, both options are good 😄

Reply:
> instead of using a discuss thread we can create a "master issue" on GitHub for feedback

Cool. Let's do that. I'll open a "feedback" issue today and update this link to point to that issue.

Reply:
> I'll open a "feedback" issue today and update this link to point to that issue.

Done in 653b5b73023a8c8459440962f39caa68a22f0623.